mantidproject/mantid
# Engineering Diffraction GUI save GSAS-II fit results to HDF5

A new algorithm, `EnggSaveGSASIIFitResultsToHDF5` (along with all the trimmings) was implemented, and integrated in the **GSAS-II Refinement** tab of the ED GUI. Now the lattice parameters, profile coefficients, fit settings used and Rwp index are saved to an HD5 file after running a refinement from the GUI.

**To test:**

Code review, make sure unit tests and documentation pass locally.

Some sample data is [here](https://github.com/mantidproject/mantid/files/1894002/GSASIIRefineFitPeaks.zip). You'll also need GSASII installed - go [here](http://docs.mantidproject.org/nightly/algorithms/GSASIIRefineFitPeaks-v1.html#installing-gsasii) for how to install GSASII.

Select the two Nexus files for **Focused Run Files** and click **Load**. Select the `.prm` file for **Instrument Parameter File**, the three `.cif`s for **Phase Files**, the location of `GSASIIscriptable.py` (part of GSASII) for **GSAS-II Installation Directory**, and choose somewhere to save out the GSAS-II project files for **New GSAS-II Project**. Fiddle with any other parameters and click **Refine All**.

Go to your output directory (on Windows `C:\EnginX_Mantid\User\<rb_number>\Runs`, on Linux `~/EnginX_Mantid/User/<rb_number>/Runs`. There should be three new `.hdf5` files, one for 280625, one for 280626 and one for both. Open them up and make sure things looks sensible - there should be a **GSAS-II Refinement** subgroup of **Bank 1** in both files. You should see datasets for **Lattice Parameters**, **Refinement Parameters** and **Rwp**.

The contents of the run number sub-groups of the file with both runs should match the contents of the corresponding individual files.

**Refinement Parameters:**

Make sure these are the same as the ones you entered into the GUI. **XMin** and **XMax** may be different - they should be the limits of the fitted data. Make sure that the two Pawley parameters were saved, if you selected **Pawley refinement**

**Profile coefficients**

If you selected to refine either Sigma or Gamma (or both), you should see whichever profile coefficients you refined in the dataset **Profile Coefficients**

Partially address #21446 

**Release Notes** 

[Here](https://github.com/mantidproject/mantid/commit/afaa018155beed151110419ea0edee0a4f8ac9ac#diff-b2bd2540c595f643521322a749fbd5dc)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
